scripts/dev.sh

```
@@ -0,0 +1,427 @@
+
+_self_test() {
+  local old_rundir="$RUNDIR"
+  RUNDIR="$(mktemp -d)"
+  trap '_stop_pidfile child; rm -f "$RUNDIR/stale.pid"; rmdir "$RUNDIR" 2>/dev/null || true' EXIT
+  setsid bash -c 'trap "" TERM; sleep 30 & wait' &
+  local leader=$!
+  _write_pidfile child "$leader"
+  [[ "$(_tracked_pid child)" == "$leader" ]]
+  _stop_pidfile child
+  wait "$leader" 2>/dev/null || true
+  ! kill -0 "$leader" 2>/dev/null
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)